### ISSUE TITLE
ProgressDialog close causes loss of focus on child control

### ISSUE THREAD

**Issue by dxdx347** [MahApps/MahApps.Metro#2532]
## What steps will reproduce this issue?

In my app, I'm loading a DocumentViewer for previewing and printing. Because loading can take up to a few seconds, I would like to display a ProgressDialog. The logic is this: display ProgressDialog, create and display DocumentViewer, close ProgressDialog. 

The problem is that when closing the dialog, the focus returns to the MainWindow, instead of remaining to the DocumentViewer window. I think it's because the dialog CloseAsync(). I've tried calling Focus(), Activate(), Show(), setting TopMost = true, setting the owner of the child window, but, although the window is initially displayed in front of the parent, the focus still returns to the parent. The only way to I restored the focus to the child was by doing a programmatic click inside the child window, in ProgressDialogController.Closed event, but the disadvantage is that there is an ugly transition (child initially appears in front, then parent, then child again).

Can CloseAsync() be avoided? Or how to close MahApps ProgressDialog without losing focus on child control?
 

### Expected outcome

The ProgressDialog should be closed without losing focus from the child window.

--
### Environment

- MahApps.Metro __v1.3.0.157__
- Windows __10__
- Visual Studio __2015__
- .NET Framework __4.6.1__

**Comment by thoemmi** [comment]
I don't think it's possible currently:
When the dialog is closed, `DialogManager.RemoveDialog` is called, and when there's no other dialog open, [it calls `MetroWindow.RestoreFocus`](https://github.com/MahApps/MahApps.Metro/blob/develop/MahApps.Metro/Controls/Dialogs/DialogManager.cs#L457). 
`RestoreFocus` now [sets the focus back](https://github.com/MahApps/MahApps.Metro/blob/develop/MahApps.Metro/Controls/MetroWindow.cs#L774-L784) to the element, which was focused when the dialog opened.

@punker76 Correct me if I'm wrong, but I guess a method like following could help. It's just an idea...
```cs
public void ResetStoredFocus()
{
    restoreFocus = null;
}
```

**Comment by punker76** [comment]
@thoemmi This is maybe a possible solution. Or an additional (public) version of the SetFocus method with a parameter which overrides the stored control...

**Comment by dxdx347** [comment]
@thoemmi Yes, it works! I tested it and focus is given to the child control.

**Comment by punker76** [comment]
@dxdx347 2 new methods now available on `MetroWindow` 12557bd6f5856831df409cc02cc59c5c61d5b506

```csharp
/// <summary>
/// Stores the given element, or the last focused element via FocusManager, for restoring the focus after closing a dialog.
/// </summary>
/// <param name="thisElement">The element which will be focused again.</param>
public void StoreFocus([CanBeNull] IInputElement thisElement = null)

/// <summary>
/// Clears the stored element which would get the focus after closing a dialog.
/// </summary>
public void ResetStoredFocus()
```

^ @thoemmi

**Comment by thoemmi** [comment]
Looks good to me 👍
